src/App.tsx

```diff
@@ -1,24 +1,286 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useState, useEffect, useMemo, useCallback,
+} from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+  updateTodoTitle,
+  updateTodoStatus,
+} from './api/todos';
 
-const USER_ID = 0;
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { getVisibleTodos, getTodosId } from './utils/utils';
+import { IdsContext } from './utils/Context/IdsContext';
+import { Options } from './types/Options';
+
+const USER_ID = 10631;
+
+export const errorMessage = {
+  forLoad: 'Unable to load todos',
+  forAdd: 'Unable to add a todo',
+  forTitle: 'the Title can not be empty',
+  forDelete: 'Unable to delete a todo',
+  forUpdate: 'Unable to update a todo',
+};
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(Options.ALL);
+  const [isError, setIsError] = useState(false);
+  const [isHidden, setIsHidden] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorText, setErrorText] = useState('');
+  const [searchValue, setSearchValue] = useState('');
+  const [deletedIds, setDeletedIds] = useState<number[]>([]);
+
+  const validValue = searchValue.trim();
+
+  const visibleTodos = useMemo(() => {
+    return getVisibleTodos(todosFromServer, filter);
+  }, [filter, todosFromServer]);
+
+  const completedTodosId = useMemo(() => {
+    return getTodosId(todosFromServer, Options.COMPLETED);
+  }, [todosFromServer]);
+
+  const notCompletedTodosId = useMemo(() => {
+    return getTodosId(todosFromServer, Options.ACTIVE);
+  }, [todosFromServer]);
+
+  const changeFilter = (value: Options) => setFilter(value);
+
+  const hideNotification = useCallback(() => {
+    setIsHidden(true);
+  }, []);
+
+  const searchHandler = (value: string) => {
+    setSearchValue(value);
+  };
+
+  const errorHandler = (text: string) => {
```

Comment:
```suggestion
  const handleSearch = (value: string) => {
    setSearchValue(value);
  };

  const handleError = (text: string) => {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364